README.md

```diff
@@ -2,31 +2,240 @@
+Unfortunately, the `HttpServletRequest` interface does not provide a method to extract a path parameter.
+However, we can use the `getRequestURI()` or `getPathInfo()` methods, along with some
+string processing, to get a path parameter value. One technique is shown below:
+
+```java
+// get the {continent_name} path parameter http://localhost:8080/continents/{continent_name}
+String continent_name = req.getRequestURI().substring("/continents/".length());
+```
+
+For simplicity, we will assume a correctly formatted URL is provided to the servlet.
+Otherwise, an exception would be thrown if the substring `/continents/` is not in the URL.
+
+
+## Servlet Implementation
+
+Let's implement a new servlet named `ContinentServlet` that will take a GET request
+of the form `http://localhost:8080/continents/{continent_name}` and return a JSON
+string containing the appropriate continent data (name, area, population).
+
+First, we need to define a class `Continent` as shown:
+
+```java
+public class Continent {
+  private String name;
+  private Integer area;
+  private Long population;
+
+  public Continent() {}
+
+  public Continent(String name, Integer area, Long population) {
+    this.name = name;
+    this.area = area;
+    this.population = population;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public Integer getArea() {
+    return area;
+  }
+
+  public void setArea(Integer area) {
+    this.area = area;
+  }
+
+  public Long getPopulation() {
+    return population;
+  }
+
+  public void setPopulation(Long population) {
+    this.population = population;
+  }
+
+  @Override
+  public String toString() {
+    return "{" +
+            "\"name\":" + "\"" + name + "\"" + "," +
+            "\"area\":" +  area +  "," +
+            "\"population\":" + population +
+            "}";
+  }
+}
+
+```
+
+The `toString()` method returns a JSON-formatted string containing to the instance variable values.
+
+We will also create a new servlet class `ContinentServlet`.  
+
+- The servlet uses a hashmap to map the continent name to a `Continent` class instance.  
+- The servlet initializes the hashmap by overriding the `init()` method,
+  which is called one time when the servlet is instantiated by the servlet container.
+- The servlet overrides the `doGet()` method to get a `Continent` based on the name specified in the path parameter.
+  - The response includes an error status if the continent is not in the hashtable.
+  - The response includes a success status if the continent is in the hashtable.
+
+```java
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
```

Comment:
I don't think this import is used? At least that's what IntelliJ told me when I imported it.

Reply:
thanks, unused import removed.